### ISSUE TITLE
Improving drafter-client API affordances

### ISSUE THREAD

**Issue by RickMoynihan** [Swirrl/drafter#452]
This is a bit of an epic issue, but I want to pull together the thinking and summarise all the issues and suggested approach in one place.

The TLDR; is that the drafter-client is currently a bit awkward to use, and we should reduce the friction a bit.

1. Figuring out how to configure it is awkward because the various configuration options aren't easily documented.  To figure it out you need to dig through various other projects for config snippets.
2. All API functions take a `token` parameter, which means clients need to handle authentication themselves.  We should hide this detail from API users in a new API namespace  See this PR #446 (which introduces the auth-provider abstraction) and the issue #453 that describes the steps required to do this.
3. Published API docs would be nice (for this and other library projects we have) See this [Design Rationale](https://www.notion.so/swirrl/Publishing-API-Documentation-for-Swirrl-s-Clojure-APIs-12a526ee248d4556884aa8f7d9eadb4f)
4. Separately having a way to document integrant components and their configuration would be very useful.  So folk can just copy/paste config into their consuming project [perhaps this future project will solve this](https://github.com/kwrooijen/keydox).

### Other affordance tweaks

- https://github.com/Swirrl/swirrl-auth0/issues/20  as a config user having to append `/api/v2` to one of the config keys 
- https://github.com/Swirrl/swirrl-auth0/issues/21 having to set the `:swagger-json` resource yourself
